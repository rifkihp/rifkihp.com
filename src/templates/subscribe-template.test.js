import React from 'react';
import renderer from 'react-test-renderer';

import SubscriberTemplate from './subscribe-template';

describe('SubscriberTemplate', () => {
  const props = {
    data: {
      site: {
        siteMetadata: {
          title: 'test',
        },
      },
    },
    pageContext: {
      updateSubscription: false,
    },
  };

  it('renders correctly when updateSubscription is false', () => {
    const tree = renderer.create(<SubscriberTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when updateSubscription is true', () => {
    const tree = renderer
      .create(<SubscriberTemplate {...{ ...props, pageContext: { updateSubscription: true } }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
