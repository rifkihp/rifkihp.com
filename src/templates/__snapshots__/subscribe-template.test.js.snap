// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriberTemplate renders correctly when updateSubscription is false 1`] = `
<StaticQuery
  render={[Function]}
/>
`;

exports[`SubscriberTemplate renders correctly when updateSubscription is true 1`] = `
<StaticQuery
  render={[Function]}
/>
`;
